### ISSUE TITLE
Accessibility and keyboard support in Cemu UI

### ISSUE THREAD

**Issue by Gamulation** [cemu-project/Cemu#614]
Hi. I am using Cemu with an accessibility program known as a Screen reader, which reads text on screen out loud. This lets me use a computer due to me being visually impaired. However. Cemu has a little problem with it. I am unable to map my controls with this. I do not use a mouse and only use a keyboard. I am able to navigate around the UI perfectly fine, but in the input settings windows is a bit weird.
I am unable to select emulated controler. It doesn't let me select one, I have no idea why.
I got a friend to select one for me, but the second issue is that I can not find the buttons to map. I can select which device to use, so that's no issue, but I can not map the actuall buttons.
I use the tab key mostly to move around and for me it seems like it skipps the hole section, but I have no idea, as I can't see it to begin with. Could these two things please be looked into?
It would make Cemu much easier to use for blind people and people who only use a keyboard for navigation.

**Comment by jn64** [comment]
Which operating system and screen reader do you use?

For the second issue, the button mapping section is not skipped. Rather, the problem is that once your focus reaches the first button's input field, pressing any key including Tab will assign it to that button, and also immediately change focus to the next button's input field.

Since Tab is a valid key for mappings, I am not sure how this should be handled. Do you know which other software handles this situation correctly?

I guess at the very least, we should make sure the button label is associated with the input field, so that the label is read when the focus changes to the input field. I don't know how this works in wxWidgets.

**Comment by Gamulation** [comment]
Hi again. I'm using latest version and the problem is still here. I am unable to select what control to emulate and not able to map controlls. Could this be fixed?
